src/Microsoft.AspNet.OData/EnableQueryAttribute.cs

```diff
@@ -71,6 +73,34 @@ public override void OnActionExecuted(HttpActionExecutedContext actionExecutedCo
                 throw Error.Argument("actionExecutedContext", SRResources.ActionExecutedContextMustHaveActionContext);
             }
 
+            bool hasExpandQueryParameter = false;
+            string queryString = request.RequestUri.Query;
+            var absoluteUri = request.RequestUri.AbsoluteUri;
+            bool queryStringIsEmpty = string.IsNullOrEmpty(queryString);
+
+            ReadOnlySpan<char> absoluteUriSpan = absoluteUri.AsSpan();
+            string baseUriWithPath = queryStringIsEmpty ? absoluteUri : absoluteUriSpan.Slice(0, absoluteUri.IndexOf('?', 0)).ToString();
+
+            if (!queryStringIsEmpty)
+            {
+                string[] queryParameters = queryString.Split('&');
+
+                hasExpandQueryParameter = queryParameters.Any(x => x.StartsWith("?$expand", StringComparison.OrdinalIgnoreCase) || x.StartsWith("$expand", StringComparison.OrdinalIgnoreCase) || x.StartsWith("expand", StringComparison.OrdinalIgnoreCase));
```

Comment:
`StartsWith` here will only check if the querystring starts with `$expand`  what if the query string has other query query options and `$expand` is at the middle or end? I think `Contains` might be a better option. Or?